### ISSUE TITLE
unhandled-error: do not report fmt.Fprintf with buffers

### ISSUE THREAD

**Issue by rostislaved** [mgechev/revive#1724]
**Description:**

There is a conflict between `revive` (`unhandled-error` rule) and `staticcheck` (`QF1012`) when using `fmt.Fprintf` with buffer types like `*strings.Builder` or `*bytes.Buffer`.

**Reproducer:**

Initial code:

```go
buf.WriteString(fmt.Sprintf("some formatted string: %v", value))
```

Running `golangci-lint --fix` rewrites it to:

```go
fmt.Fprintf(&buf, "some formatted string: %v", value)
```

This satisfies `staticcheck` (`QF1012`), which prefers `fmt.Fprintf` over `WriteString(fmt.Sprintf(...))`.

However, after this rewrite, `revive` reports:

```text
unhandled-error: Unhandled error in call to function fmt.Fprintf
```

**Problem:**

`fmt.Fprintf` returns an `error`, but in this case it writes into:

* `*strings.Builder`
* `*bytes.Buffer`

Both implement `Write` methods that never return a non-nil error. Therefore, checking the returned error is unnecessary and adds noise.

This creates a situation where:

* `staticcheck` enforces `fmt.Fprintf`
* `revive` enforces checking the returned error
* the error is guaranteed to be `nil` in practice

**Current workarounds:**

1. Add redundant error handling:

```go
_, err = fmt.Fprintf(&buf, "...")
if err != nil {
    return err
}
```

This introduces unnecessary boilerplate and may force API changes.

2. Add `//nolint:revive`:

```go
//nolint:revive // strings.Builder.Write always returns nil
fmt.Fprintf(&buf, "...")
```

This works but requires manual intervention after `--fix`.

**Expected behavior:**

`revive` should not report `unhandled-error` when `fmt.Fprintf` writes to known safe writers like:

* `*strings.Builder`
* `*bytes.Buffer`

Alternatively, allow configuration of `unhandled-error` rule to ignore cases based on argument types (not only function names).
